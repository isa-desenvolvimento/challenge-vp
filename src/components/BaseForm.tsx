import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
