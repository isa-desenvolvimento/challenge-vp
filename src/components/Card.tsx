import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'

export const Card = ({
  header,
  content,
  footer,
  primary,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer role="card">
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
