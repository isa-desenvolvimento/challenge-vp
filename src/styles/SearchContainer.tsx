import styled from 'styled-components'

export const StyleSearchContainer = styled.div`
  background: ${({ theme }) => theme.pallet.white};
  display: grid;
  grid-template-columns: 30% 20% auto;
  justify-content: unset;
  align-items: baseline;

  grid-gap: 1rem;

  @media screen and (max-width: 600px) {
    display: revert;
  }
`

export const StyleButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`
