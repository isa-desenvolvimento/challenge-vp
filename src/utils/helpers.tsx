import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
