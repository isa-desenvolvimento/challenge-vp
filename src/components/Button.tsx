import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
