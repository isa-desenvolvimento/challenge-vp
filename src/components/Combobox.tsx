import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
