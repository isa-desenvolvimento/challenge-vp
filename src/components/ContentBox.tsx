import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
