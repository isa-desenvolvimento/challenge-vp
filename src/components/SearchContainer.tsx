import {
  StyleSearchContainer,
  StyleButtonContainer,
} from '@/styles/SearchContainer'
import { Button } from '@/components/Button'
import { Combobox } from '@/components/Combobox'
import { InputComponent } from '@/components/InputComponent'
import { BaseForm } from './BaseForm'
import { SearchContainerProps } from '@/types/global'

export const SearchContainer = ({
  options,
  onSubmit,
}: SearchContainerProps) => {
  const renderBaseForm = () => (
    <StyleSearchContainer role="searchContainer">
      <InputComponent
        name="search"
        iconProps="search"
        placeholder="Pesquisa Livre"
      ></InputComponent>
      <Combobox options={options} placeholder="Status" name="status" isFormik />
      <StyleButtonContainer>
        <Button
          type="submit"
          primary
          onClick={() => {
            console.log
          }}
        >
          Buscar
        </Button>
      </StyleButtonContainer>
    </StyleSearchContainer>
  )

  return (
    <BaseForm
      baseForm={renderBaseForm()}
      onSubmit={onSubmit}
      initialValues={{
        search: '',
        status: '',
      }}
    />
  )
}
