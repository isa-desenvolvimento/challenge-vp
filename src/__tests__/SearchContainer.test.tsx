import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { SearchContainer } from '@/components/SearchContainer'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

describe('<SearchContainer />', () => {
  const options = [
    { value: 'Done', id: 1 },
    { value: 'Warning', id: 2 },
    { value: 'Error', id: 3 },
  ]
  const onSubmit = jest.fn()

  jest.mock('@fortawesome/free-solid-svg-icons')
  jest.mock('@fortawesome/fontawesome-svg-core')

  beforeEach(() =>
    renderWithProvider(
      <SearchContainer onSubmit={onSubmit} options={options} />
    )
  )

  it('should render text SearchContainer', () => {
    expect(screen.getByRole(/searchContainer/i).textContent).toContain('Buscar')
  })

  it('should render snapshot', () => {
    expect(screen.getByRole(/searchContainer/i)).toMatchSnapshot()
  })
})
