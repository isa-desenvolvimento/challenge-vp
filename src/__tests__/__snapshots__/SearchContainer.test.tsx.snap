// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchContainer /> should render snapshot 1`] = `
.c0 {
  background: #ffff;
  display: grid;
  grid-template-columns: 30% 20% auto;
  -webkit-box-pack: unset;
  -webkit-justify-content: unset;
  -ms-flex-pack: unset;
  justify-content: unset;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  grid-gap: 1rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 {
  background: #36a4a8;
  color: #ffff;
  font-size: 1em;
  margin: 1em;
  border: none;
  border-radius: 4px;
  padding: 0.8em 2em;
  width: auto;
  text-transform: uppercase;
  text-overflow: ellipsis;
  cursor: pointer;
}

.c7:hover {
  opacity: 0.8;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  gap: 1rem;
  padding: 1rem;
  text-overflow: ellipsis;
  background: #eceff5;
  color: #3AA3AA;
  border: none;
  border-radius: 4px;
  width: auto;
}

.c5 {
  background: #eceff5;
  color: #3AA3AA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 1rem;
  padding: 1rem;
  text-overflow: ellipsis;
  background: #eceff5;
  color: #3AA3AA;
  border: none;
  border-radius: 4px;
  width: auto;
  margin: 1rem;
}

.c2 {
  padding: 0;
  margin: 0;
  color: #3AA3AA;
  background: none;
  border: none;
  text-overflow: ellipsis;
}

@media screen and (max-width:600px) {
  .c0 {
    display: revert;
  }
}

@media screen and (max-width:600px) {
  .c7 {
    width: 100%;
  }
}

@media screen and (max-width:600px) {
  .c4 {
    margin: 1rem;
  }
}

<div
  class="c0"
  role="searchContainer"
>
  <label
    class="InputComponent__StyleContainer-sc-o1q8z5-0"
    role="input-component"
  >
    <div
      class="c1"
      name="search"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="icon"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
      <input
        class="c2"
        name="search"
        placeholder="Pesquisa Livre"
        role="search"
        value=""
      />
    </div>
  </label>
  <div
    class="c3"
    role="combobox"
  >
    <select
      class="c4"
      name="status"
      role="combobox"
    >
      <option
        class="c5"
        name="status.defaultValue"
        value=""
      >
        Status
      </option>
      <option
        class="c5"
        name="status.Done"
        value="Done"
      >
        Done
      </option>
      <option
        class="c5"
        name="status.Warning"
        value="Warning"
      >
        Warning
      </option>
      <option
        class="c5"
        name="status.Error"
        value="Error"
      >
        Error
      </option>
    </select>
  </div>
  <div
    class="c6"
  >
    <button
      class="c7"
      role="button"
      type="submit"
    >
      Buscar
    </button>
  </div>
</div>
`;
